ops/testing.py

```diff
@@ -1080,6 +1094,17 @@ def get_pebble(self, socket_path: str):
         return client
 
     def planned_units(self):
+        """Simulate processing the return from goal-state into a count of planned units.
+
+        If self._planned_units is None, then we simulate what the Juju controller will do, which is
+        to report the number of peers, plus one (we include this unit in the count).
+
+        For testing purposes, a charm author can set self._planned_units explicitly by calling
+        "Harness.set_planned_units".
```

Comment:
```suggestion
        If self._planned_units is None, then we simulate what the Juju controller will do, which is
        to report the number of peers, plus one (we include this unit in the count). This can be
        overridden for testing purposes: a charm author can set the number of planned units 
        explicitly by calling `Harness.set_planned_units`
```